### DISCUSSION TITLE
Lesson 9 (6:57)

### DISCUSSION THREAD

**Discussion by DanCsoftware** [Cyfrin/foundry-full-course-cu#1545] [General]
Is it normal for the request on the VRFCoordinator side to take this long? 

I was following along the course and everything I was doing so far was pretty much reflecting what Patrick was doing. With the exception of him having a different subId than I had. 

<img width="1258" alt="Screenshot 2024-03-29 at 12 12 27 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/35975936/916e2160-74c8-4084-a4a8-e91c90f32c48">

It does also look like the requestUpkeep did make a transaction with chainlink because the link balance seemed to have been taken.

<img width="1244" alt="Screenshot 2024-03-29 at 12 16 24 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/35975936/0abf770a-92df-4a5b-a578-5fa314bea446">

If this is normal behavior and I just need to wait longer please disregard, but just wanted to be sure I didn't make an error from anywhere.

**Comment by EngrPips** [comment]
Hello @DanCsoftware , I have seen few people run into this issue of needing to wait for a long time to get back a random number which is totally unusual and abnormal but till now i haven't been able to figure out why that happens. i would say maybe browse google for this issue or ethereum stack exchange to see if people have reported such issue concerning chainlink VRF and how they solved the issue
